infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -96,8 +96,11 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
             // Check that all opaque entries contain the correct streams
             for (OpaqueEntry opaqueEntry : newOpaqueEntryList) {
                 if (!streamMap.keySet().containsAll(opaqueEntry.getEntries().keySet())) {
-                    log.error("txMessage contains noisy streams {}, expecting {}", opaqueEntry.getEntries().keySet(), streamMap);
-                    throw new ReplicationWriterException("Wrong streams set");
+                    // For version compatibility concern, we only log warning messages for noisy streams without throwing an exception.
+                    // Because during rolling upgrade the Active and Standby might have different set of streams to replicate, we
+                    // should ignore those streams sent by Active but not part of the set of streams on Standby.
+                    log.warn("txMessage contains noisy streams {}, expecting {}. Note that noisy streams are allowed during rolling upgrade." +
```

Comment:
This message and this whole 'if' statement, doesn't seem to be anymore needed, right? cause you print this info in line 112 and now we don't throw an error...

Reply:
Yes correct... they are basically the same